packages/grafana-ui/src/components/Combobox/Combobox.tsx

```diff
@@ -0,0 +1,65 @@
+import { useCombobox } from 'downshift';
+import React, { useState } from 'react';
+
+import { SelectableValue } from '@grafana/data';
+
+import { Icon } from '../Icon/Icon';
+import { Input, Props as InputProps } from '../Input/Input';
+
+type Value = string | number;
+type Option = {
+  label: string;
+  value: Value;
+};
+
+interface ComboboxProps
+  extends Omit<InputProps, 'width' | 'prefix' | 'suffix' | 'value' | 'addonBefore' | 'addonAfter' | 'onChange'> {
+  onChange: (val: Option) => void;
+  value: Value;
+  options: Option[];
+}
+
+function itemToString(item: Option | null) {
+  return item?.label || '';
+}
+
+function itemFilter(inputValue: string) {
+  const lowerCasedInputValue = inputValue.toLowerCase();
+
+  return (item: Option) => {
+    return (
+      !inputValue ||
+      item?.label?.toLowerCase().includes(lowerCasedInputValue) ||
+      item?.value?.toString().toLowerCase().includes(lowerCasedInputValue)
+    );
+  };
+}
+
+export const Combobox = ({ options, onChange, value, ...restProps }: ComboboxProps) => {
+  const [items, setItems] = useState(options);
+
+  const { getInputProps, getMenuProps, getItemProps, isOpen } = useCombobox({
+    items,
+    itemToString,
+    selectedItem: options.find((option) => option.value === value) || null,
```

Comment:
@joshhunt This is the side effect of not passing the object as a value

Reply:
Nice. Lets memoize it! 👍

Reply:
Nice! Lets `useMemo` it as well to make sure we don't loop over more than we need to!

Reply:
@dprokop The multi select will initially be its own thing, then we'll look at if we can extract anything.